design-patterns/domain-driven-design/pattern1/README.md

```diff
@@ -0,0 +1,194 @@
+                               "type": "string"
+                           }
+                       }
+                   }
+```
+
+The required constraint on the `customerId`, `firstname`, `lastName`, and on the next complex type `address`, and property of the `address` i.e `zipcode`.
+
+
+```json
+"required": [
+                    "customerId",
+                    "firstName",
+                    "lastName",
+                    "address"
+                ]
+```
+
+4. Press “Create”. Wait for 5 minutes for Share to finish Uni provisioning.
+
+After Uni provisioning is complete, you will notice that your Uni will appear in the `RUNNING` status on the home page. 
+	
+## Explore the Uni
+
+Click the Uni you just created. Click on the `Entity Explorer`. You will notice the `SchemaOwner` and the `CustomerAccount` entity on the left. 
+	
+`SchemaOwner` is a Value Object; it’s a singleton value object with no unique identifier. You will use it to keep track of who is the Uni `owner` along with the `emailId` of Uni owner. It is good to have this information in the Uni itself. Click `Edit` and fill the `group` and the `emailId` properties.
+
+Let’s create a customer record. Click `CustomerAccount` entity. Click `Create Customer Account`. Enter customer account details. For `CustomerId`, pick a number. I use the [10001,10002…] series for testing. In the real world, either you will have your own sequence generator for `customerId`, or you will use the `customerId` from different enterprise source of truth system.
+
+You will see all the customer account records you created under the `CustomerAccount` entity. You can edit a specific record, change address, and in the version history, you will see both the current and the previous state of the customer account you just edited. This is the ledger aspect of the Uni. Uni keeps both the current and previous versions of the record. 
+
+What you did do using `Entity Explorer`. You can do the same through `GraphQL Explorer`. You don’t have to type every single command and attribute to write a GraphQL query. You can build a GraphQL command by selecting the commands and the attributes in the `Schema Explorer`.
+
+Let’s create a new customer record and see whether our required constraint on `customerId`,`firstname`,`lastname`, `address{zipcode}` is working or not?
+
+Navigate to `GraphQL Explorer`, it has three sections, the left section is GraphQL commands for your Uni, you can use it to build a new query or mutation. In the middle section you will write your query and press the green button on the top to run it. And, you will see the results(success/error) of the query on the right-hand section. 
+
+Try following queries in the `GraphQL Explorer` and note the difference between results. 
+
+Mutation with required `zipcode` property. 
+
+
+```graphql
+mutation m{
+  add_CustomerAccount_async(input: {customerId: "10001", firstName: "Jay", lastName: "Pipes"}) {
+    error
+  }
+}
```

Comment:
And continues down to here

Reply:
I am not seeing this issue in my editor. The preview also looks fine.

Reply:
resolved